qutebrowser/mainwindow/mainwindow.py

```diff
@@ -483,9 +486,12 @@ def _connect_signals(self):
     @pyqtSlot(bool)
     def _on_fullscreen_requested(self, on):
         if on:
+            self._state_before_fullscreen = self.windowState()
             self.showFullScreen()
         else:
-            self.showNormal()
+            self.setWindowState(self._state_before_fullscreen)
+        log.misc.debug('on: {}, state before fullscreen: {}'
+                       .format(on, self._state_before_fullscreen))
```

Comment:
Does this log something useful? If not, might want to use `debug.qflags_key` (from `qutebrowser.utils.debug`).